lib/jekyll-admin/server.rb

```diff
@@ -75,7 +75,11 @@ def write_file(path, content)
       File.open(path, "wb") do |file|
         file.write(content)
       end
-      site.process
+      if ENV["RACK_ENV"]
+        site.process
+      else
+        site.read
+      end
```

Comment:
> Will that introduce a race condition if the file isn't yet written to disk?

Yeah, I encountered that once but comparing with the current version, this is an improvement 😄